### PR TITLE
live_migration: Add new case about suspend virsh client

### PR DESCRIPTION
**Pull request by cliping**
RHEL-120213 - [Migration] Suspend virsh during live migration (p2p
migration)

Signed-off-by: lcheng <lcheng@redhat.com>

### CODE REVIEW

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --vt-machine-type q35 virsh.live_migration.suspend_virsh.p2p_live.without_postcopy --vt-connect-uri qemu:///system
WARNING:root:No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
WARNING:root:No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : 47dc08b36768f57706e57033579e5dc2a6d6cec9
JOB LOG    : /root/avocado/job-results/job-2021-09-03T03.40-47dc08b/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.live_migration.suspend_virsh.p2p_live.without_postcopy: PASS (132.02 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 132.57 s

**Review by kamilvarga — APPROVED** [review]
LGTM.

File: provider/migration/migration_base.py [hunk 1]
```diff
@@ -192,3 +193,20 @@ def check_event_output(params, test, virsh_session=None, remote_virsh_session=No
     if expected_event_target and remote_virsh_session:
         target_output = remote_virsh_session.get_stripped_output()
         check_output(target_output, eval(expected_event_target), test)
+
+
+def suspend_virsh(params):
```
Comment (chunfuwen) [inline provider/migration/migration_base.py:198]: a little confused about the method name:suspend_virsh.
Check implementation part of this method, it sounds like control context switch between postcopy and precopy.
Maybe the method name can be changed to more proper one, e.g: switch_between_postcopy_and_precopy()

File: provider/migration/migration_base.py [hunk 2]
```diff
@@ -192,3 +193,20 @@ def check_event_output(params, test, virsh_session=None, remote_virsh_session=No
...
+
+    :param params: dict, used to setup the connection
+
+    """
+    vm_name = params.get("migrate_main_vm")
+    migrate_speed_high = params.get("migrate_speed_high")
+    postcopy_options = params.get("postcopy_options")
+    migration_test = params.get("migration_test")
+
+    migration_test.do_cancel(sig=signal.SIGTSTP)
```
Comment (Yingshun) [inline provider/migration/migration_base.py:210]: I think this function may be related to a special case only, I don't think I see the possibility of calling this function in other cases. Sorry, please correct me if I'm wrong.
`action_during_mig` supports a List value, so we can  do some operations in order. How about setting the 1st item of  `action_during_mig` to a value that can execute do_cancel(), and the 2nd one(maybe you need to create a new function to handle 3 args) to set high migration speed?

File: provider/migration/migration_base.py (hunk 1 above)
Comment (cliping) [reply provider/migration/migration_base.py]: Updated. Thanks.

File: provider/migration/migration_base.py (hunk 2 above)
Comment (cliping) [reply provider/migration/migration_base.py]: Updated, thanks.

File: provider/migration/migration_base.py [hunk 3]
```diff
@@ -205,3 +207,18 @@ def poweroff_src_vm(params):
...
+def set_migrate_speed_to_high(params):
+    """
+    Set migrate speed to high value
+
+    :param params: dict, used to setup the connection
+    """
+    vm_name = params.get("migrate_main_vm")
+    migrate_speed_high = params.get("migrate_speed_high")
+    postcopy_options = params.get("postcopy_options")
+    migration_test = params.get("migration_test")
```
Comment (Yingshun) [inline provider/migration/migration_base.py:221]: I found you have imported 'MigrationTest' in L10 with pylint disabled. Hmm, could you please help to check if the 'migration_test' object must be passed through 'params'? Thank you!

File: provider/migration/migration_base.py [hunk 4]
```diff
@@ -1,11 +1,13 @@
 import logging
 import types
 import re
+import signal                                        # pylint: disable=W0611
```
Comment (Yingshun) [inline provider/migration/migration_base.py:4]: Can we remove it?

File: provider/migration/migration_base.py (hunk 3 above)
Comment (cliping) [reply provider/migration/migration_base.py]: Updated, thanks.

File: provider/migration/migration_base.py (hunk 4 above)
Comment (cliping) [reply provider/migration/migration_base.py:4]: No. If remove it, we will get following error.


2021-09-17 06:04:04,752 stacktrace       L0045 ERROR|   File "/var/lib/avocado/data/avocado-vt/virttest/test-providers.d/downloads/io-github-autotest-libvirt/provider/migration/migration_base.py", line 49, in parse_funcs
2021-09-17 06:04:04,752 stacktrace       L0045 ERROR|     func_param = eval(func_param)
2021-09-17 06:04:04,752 stacktrace       L0045 ERROR|   File "<string>", line 1, in <module>
2021-09-17 06:04:04,752 stacktrace       L0045 ERROR| NameError: name 'signal' is not defined

**Review by Yingshun — APPROVED** [review]

File: provider/migration/migration_base.py [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
 
 from virttest import virsh                           # pylint: disable=W0611
 
 from virttest.utils_conn import TLSConnection
 from virttest.utils_libvirt import libvirt_network   # pylint: disable=W0611
+from virttest.migration import MigrationTest         # pylint: disable=W0611
```
Comment (Yingshun) [inline provider/migration/migration_base.py:10]: can we remove '# pylint: disable=W0611'?

Comment (cliping) [reply provider/migration/migration_base.py:10]: If remove '# pylint: disable=W0611', run inspekt will fail.

Global check FAIL
************* Module migration_base
W0611:  4,0: : Unused import signal
W0611: 10,0: : Unused MigrationTest imported from virttest.migration
Error: Process completed with exit code 1.

Comment (Yingshun) [reply provider/migration/migration_base.py:10]: It's strange, I applied this pr and removed '# pylint: disable=W0611', inspekt check passed:

Comment (cliping) [reply provider/migration/migration_base.py:10]: Error information:

https://github.com/autotest/tp-libvirt/runs/3623194691?check_suite_focus=true

Comment (cliping) [reply provider/migration/migration_base.py]: Sorry, I made a mistake. Updated again. Thanks.

**Review by Yingshun — APPROVED** [review]

**Review by fangge1212 — APPROVED** [review]

**Review by chunfuwen — APPROVED** [review]
LGTM

**Pull request merged**